### ISSUE TITLE
5V VS 3.3V

### ISSUE THREAD

**Issue by MikeA1s** [snakeye/ch340g-esp32#2]
Me thinks this schematics is wrong. Me thinks this is more accurarte.
https://pcbartists.com/design/embedded/esp32-ch340-programmer-schematic/

**Comment by snakeye** [comment]
On this schematics, the CH340G is working in 5V mode. I would expect 5V on the TX line as well as on EN and IO lines. The ESP32 won't die probably but it does not look good to me.

**Comment by MikeA1s** [comment]
yes and thank you... but you are exceeding 3,3v on data pins where data sheet says "circuit voltage which is
connected with CH340 is no more than 3.3V" if on 3.3v. Nodemcu has it connected to 5v too? https://github.com/nodemcu/nodemcu-devkit
I am just trying to get optimal stuff here, don't mind me😁

**Comment by snakeye** [comment]
I'm not getting the idea about 5V in the circuit. Everything is powered with 3.3V
